files/class-vip-filesystem.php

```diff
@@ -385,4 +418,94 @@ public function filter_get_attached_file( $file, $attachment_id ) {
 
 		return $file;
 	}
+
+	/**
+	 * Filter `cron_schedules` output
+	 *
+	 * Add a custom schedule for a 5 minute interval
+	 *
+	 * @param   array   $schedule
+	 *
+	 * @return  mixed
+	 */
+	public function filter_cron_schedules( $schedule ) {
+		if ( isset( $schedule[ 'vip_five_minutes' ] ) ) {
+			return $schedule;
+		}
+
+		$schedule['vip_five_minutes'] = [
+			'interval' => 300,
+			'display' => __( 'Once every 5 minutes' ),
+		];
+
+		return $schedule;
+	}
+
+	public function schedule_update_job() {
+		if ( get_option( self::OPT_ALL_FILESIZE_PROCESSED ) ) {
+			if ( wp_next_scheduled( self::CRON_EVENT_NAME ) ) {
+				wp_clear_scheduled_hook( self::CRON_EVENT_NAME );
+			}
+
+			return;
+		}
+
+		if (! wp_next_scheduled ( self::CRON_EVENT_NAME )) {
+			wp_schedule_event(time(), 'vip_five_minutes', self::CRON_EVENT_NAME );
+		}
+
+		add_action( self::CRON_EVENT_NAME, [ $this, 'update_attachment_meta' ] );
+	}
+
+	/**
+	 * Cron job to update attachment metadata with file size
+	 */
+	public function update_attachment_meta() {
+		trigger_error(
+			sprintf( 'Starting %s... $vip-go-streams-debug', self::CRON_EVENT_NAME ),
+			E_USER_NOTICE );
+		if ( get_option( self::OPT_ALL_FILESIZE_PROCESSED ) ) {
+			// already done. Nothing to update
+			trigger_error(
+				sprintf( 'Already completed. Exiting %s... $vip-go-streams-debug', self::CRON_EVENT_NAME ),
+				E_USER_NOTICE );
+			return;
+		}
+
+		$updater = new Meta_Updater( 1000 );
+
+		$max_id = (int) get_option( self::OPT_MAX_POST_ID );
+		if ( ! $max_id ) {
+			$max_id = $updater->get_max_id();
+			update_option( self::OPT_MAX_POST_ID, $max_id, false );
+		}
+
+		$start_index = get_option( self::OPT_NEXT_FILESIZE_INDEX, 0 );
+		$end_index = $start_index + $updater->get_batch_size();
+
+		do {
+			if ( $start_index > $max_id ) {
+				// This means all attachments have been processed so marking as done
+				update_option( self::OPT_ALL_FILESIZE_PROCESSED, 1 );
+
+				return;
+			}
+
+			$attachments = $updater->get_attachments( $start_index, $end_index );
+
+			$start_index = $end_index + 1;
+			$end_index = $start_index + $updater->get_batch_size();
+		} while ( empty( $attachments ) );
+
+		if ( $attachments ) {
+			$updater->update_attachments( $attachments );
+		}
+
+		// All done, update next index option
+		trigger_error(
+			sprintf( 'Batch %d to %d completed. Updating options... $vip-go-streams-debug',
+				$start_index, $end_index ),
```

Comment:
Minor: Just realized, now that we have the loop, we could end up with a gap in these messages. Maybe we should have an `original_start_index` variable that we would use here instead?

Reply:
Yeah I noticed that too in the logs. I'll change it along with sending the messages to IRC